commit deefaef587ed3bee2706cb6e53ee3468e682932e (refs/changes/64/4064/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-05-31T14:19:03-07:00 (1 year, 4 months ago)
    
    TRITON-457 sdc-scripts update in TRITON-380 broke cns and sdc-manatee
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/lib/util.sh b/lib/util.sh
index 4ff5a1e..782e28d 100644
--- a/lib/util.sh
+++ b/lib/util.sh
@@ -588,13 +588,23 @@ function sdc_log_rotation_setup_end
     fi
 
     #
-    # Scrub existing logadm(1M) invocations from the root crontab:
+    # Scrub existing logadm(1M) invocations from the root crontab. Explictly
+    # we want to ensure we scrub:
+    #
+    # 1. entries in our base images, which are various of the form:
+    #       10 0 * * * /usr/sbin/logadm
+    #       0 * * * * /usr/sbin/logadm
+    #       # Rotate system logs
+    #       10 3 * * * /usr/sbin/logadm
+    # 2. the entry we are adding below (to support this function being
+    #    re-runnable)
+    #       0 * * * * /usr/sbin/logadm -v >> /var/log/logadm.log 2>&1
     #
     if ! crontab=$(/usr/bin/crontab -l); then
         fatal "could not read root crontab"
     fi
     if ! crontab=$(/usr/bin/sed -e '/# Rotate system logs/d' \
-      -e '/\/usr\/sbin\/logadm$/d' <<< "$crontab"); then
+      -e '/ \/usr\/sbin\/logadm/d' <<< "$crontab"); then
         fatal "could not remove logadm(1M) entries from crontab"
     fi
     if [[ $crontab =~ $logadm_regex ]]; then
